Lib/gzip.py

```diff
@@ -338,6 +338,20 @@ def read1(self, size=-1):
             size = io.DEFAULT_BUFFER_SIZE
         return self._buffer.read1(size)
 
+    def readinto(self, b):
+        self._check_not_closed()
```

Comment:
I think it would be good to make a new helper for "check not closed and check READ", call it in the 5 spots where these 4 lines are copied currently (Passing in the function name)

Reply:
Done.